tests/c/trace_too_long.c

```diff
@@ -0,0 +1,46 @@
+// # This test can be very slow, so ignore it except where we know it'll run
+// # fast enough.
+// ignore-if: test "$YKB_TRACER" != "hwt"
+// Run-time:
+//   env-var: YKD_SERIALISE_COMPILATION=1
+//   env-var: YKD_PRINT_JITSTATE=1
+//   stderr:
+//     ...
+//     jit-state: stop-tracing
+//     jit-state: trace-compilation-aborted: Trace too long
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <yk.h>
+#include <yk_testing.h>
+
+int main(int argc, char **argv) {
+  YkMT *mt = yk_mt_new(NULL);
+  yk_mt_hot_threshold_set(mt, 0);
+  YkLocation loc1 = yk_location_new();
+  YkLocation loc2 = yk_location_new();
+
+  int i = 2000;
+  NOOPT_VAL(loc1);
+  NOOPT_VAL(loc2);
+  NOOPT_VAL(i);
+  NOOPT_VAL(i);
+  YkLocation *loc = &loc1;
+  while (i > 0) {
+    yk_mt_control_point(mt, loc);
+    if (i == 2000)
```

Comment:
This is an unusual use of the JIT, but I guess there's no reason why it shouldn't be allowed...

Reply:
I don't think "unusual" makes sense in the context of a test: this is a way of determinstically emulating an "unfortunate" trace.

Reply:
What I don't get is: if the intention of the test is to make a really long trace, why do we need two meta-tracers?

Reply:
Did you notice the subtle difference in the way the tests signify failure? PT signifies failure in `stop`, hwt mapping in "mapping". We *could* unify those two things, but that would be more work. So we currently need (and needed before this PR!) two separate tests.

Reply:
My apologies, I wasn't clear.

What I meant to say is: why do you need two *locations* to make a long trace? I'm not saying you don't, but it isn't obvious to me.

Reply:
You need to start tracing and then not have it stop for ages. You can do this in various ways: I randomly chose to do it in this way, in part because I don't think the optimiser can easily worm around this. That said, I wouldn't read too much into it other than "the test does test the thing we want".